### PR TITLE
Update documentation with new tutorial about working with misaligned data

### LINKED ISSUE #277: Update documentation after "Handling unaligned data" track
### 🚀 Feature Request

We should update our documentation to reflect new functionality.

### Proposal

- Update glossary
  - Timestamp
  - Regular timestamp
  - Irregular timestamp
  - Alignment
  - Aligned data
  - Misaligned data
- Add new tutorial into intermediate section (or is it advanced?)
  - Loading data
  - Theory: int timestamp, unaligned data which can be modeled using external timestamp
  - Preparing data using utils
  - Creating `TSDataset` using new constructor
  - Example of pipeline (with timestamp transforms)
  - Note that it could also be used with irregular data at user's own risk

### Test cases

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by d-a-bunin**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look #277.

## Closing issues
Closes #277.

### CODE REVIEW

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]

File: docs/source/glossary.rst [hunk 1]
```diff
@@ -10,10 +10,19 @@ This page lists some common terms used in documentation of the library.
    Time series
```
[the remark this answers is not in the document]
Comment (d-a-bunin) [reply docs/source/glossary.rst:10]: I tried to order them by their meaning.

File: docs/source/glossary.rst [hunk 2]
```diff
@@ -29,7 +38,14 @@ This page lists some common terms used in documentation of the library.
    Hierarchical time series
       Multiple :term:`time series` having a level structure in which higher levels can be disaggregated
       by different attributes of interest into series of lower levels.
-      See :doc:`tutorials/14-hierarchical_pipeline`.
+      See :doc:`tutorials/303-hierarchical_pipeline`.
```
[the remark this answers is not in the document]
Comment (d-a-bunin) [reply docs/source/glossary.rst:41]: That is all I was able to find with search.

**Comment by d-a-bunin** [comment]
<p>These external timestamps are sitting inside <code>df_exog</code>, they are limmited by the size of <code>df_exog</code>. If we want to extend them into infinity we need a different logic.  </p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846798'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by d-a-bunin** [comment]
<p>You are forecasting the values with integer index, original index is just a feature.</p><p>If you have this feature in your forecast you know which value correspond to which timestamp.</p><p>If you don't have this feature in your forecast you could use make_timestamp_df_from_alignment to recreate it, I think. I haven't really thought about this problem. Do you have any ideas how should it work for the user?</p><p> </p><p>No, currently we can't draw forecasts with original timestamps.</p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846896'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by d-a-bunin** [comment]
<ul><li>aligned + regular: support natively</li><li>aligned + irregular: support* using int index + external timestamp, but we don't have any guarantees that it will work fine</li><li>misaligned + regular: support using int index + external timestamp</li><li>misaligned + irregular: support* using int index + external timestamp, but we don't have any guarantees that it will work fine</li></ul><p> </p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846733'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by d-a-bunin** [comment]
<p>I don't really get what do you mean under inference strategies here, can you elaborate?</p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846714'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by alex-hse-repository** [comment]
<p>For example we can fit the pipeline on misaligned data and run inference on aligned dataset</p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846714'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by alex-hse-repository** [comment]
<p>Is it helpful info, how do you think?</p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846733'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by alex-hse-repository** [comment]
<p>Looks like parameter which I always set to 100000</p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846798'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by alex-hse-repository** [comment]
<ol><li>I don't know the use case of forecasts with misaligned data, I thought we did this track to handle cases where we wan to train on misaligned data and forecast on aligned data</li><li>May be we can add such functionality(about plotting forecasts) -- but not sure, see (1)</li></ol>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846896'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]
See comments

**Comment by d-a-bunin** [comment]
<p>It doesn't work that simple. Working with misaligned data is implemented by working with integer timestamp, that's all.</p><p> </p><p>You can fit on data with integer timestamp where some segments are misaligned to others (e.g. they are really old) and later make an inference on subset of segments that you want to forecast.</p><p> </p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846714'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by d-a-bunin** [comment]
<p>If you have a lot of memory you are free to go)</p><p> </p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846798'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by d-a-bunin** [comment]
<p>Let's assume we have a data there some segments are really old and doesn't have to be forecasted anymore. We can align them with new segments and fit our pipeline. After that we can use that pipeline to forecast only recent segments that we are working with.</p><p>In that scenario we were able to use the patterns that we learnt from old segments.</p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846896'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by alex-hse-repository** [comment]
<p>Let's add example for this scenario "pipeline on misaligned data and run inference on aligned dataset"</p>

---

 <a href='https://app.reviewnb.com/etna-team/etna/pull/288/discussion/#comment-2024846714'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by alex-hse-repository — APPROVED** [review]

**Pull request merged**